doc/setup.md

```diff
@@ -1,46 +1,122 @@
+This document instructs you on how to setup your development environment and run Nanvix.
 
 > ⚠️ These instructions assume that you have superuser privileges on the machine.
 
-## 1. Clone this Repository
+## 1. Clone this Repository and Submodules
 
 ```bash
-export WORKSPACE=$HOME/nanvix                   # Change this if you want.
-mkdir -p $WORKDIR                               # Create workspace.
-cd $WORKDIR                                     # Switch to workspace.
-git clone https://github.com/nanvix/nanvix.git  # Clone repository.
-cd nanvix                                       # Switch to source tree.
+export WORKDIR=$HOME/nanvix                                          # Change this if you want.
+mkdir -p $WORKDIR                                                    # Create workspace.
+cd $WORKDIR                                                          # Switch to workspace.
+git clone --recurse-submodules https://github.com/nanvix/nanvix.git  # Clone repository.
+cd nanvix                                                            # Switch to nanvix source tree.
 ```
 
 ## 2. Install Dependencies
 
+> ⚠️ For this step, case you will build QEMU, remove QEMU from shell scripts.
+
+### Ubuntu 22.04
+```bash
+cat build/scripts/setup/ubuntu.sh        # Inspect what is going to be installed.
+sudo -E ./build/scripts/setup/ubuntu.sh  # Install dependencies.
+```
+
+### Arch Linux
 ```bash
-cat scripts/setup/ubuntu.sh            # Inspect what is going to be installed.
-sudo -E ./scripts/setup/ubuntu.sh x86  # Install dependencies.
+cat build/scripts/setup/arch.sh        # Inspect what is going to be installed.
+sudo -E ./build/scripts/setup/arch.sh  # Install dependencies.
+```
 
-# Install Rust toolchain.
+### Install Rust toolchain
+``` bash
 curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+exec $SHELL  # Restart shell to update path.
 rustup component add rust-src
 ```
 
-## 3. Build Toolchain
+## Optional
+
+> ⚠️ Next instructions are necessary only if you want compile QEMU and C language toolchain.
+
+### Build Toolchain
 
 > ⚠️ This step may take some time to complete.
 
 ```bash
-export TARGET=x86                     # Select x86 as your target architecture.
-./scripts/setup/toolchain.sh $TARGET  # Build GCC, Binutils, and GDB.
+export TARGET=x86                           # Select x86 as your target architecture.
+./build/scripts/setup/toolchain.sh $TARGET  # Build GCC, Binutils, and GDB.
 ```
 
-## 4. Build QEMU
+### Build QEMU
 
 > ⚠️ This step may take some time to complete.
 
 ```bash
-export TARGET=x86                # Select x86 as your target architecture.
-./scripts/setup/qemu.sh $TARGET  # Build QEMU.
+export TARGET=x86                      # Select x86 as your target architecture.
+./build/scripts/setup/qemu.sh $TARGET  # Build QEMU.
 ```
+# Run Nanvix
+
+This section instructs you on how to compile and run Nanvix. 
+
+## Using Microvm
+
+Compile Microvm:
+
+```bash
+cd $WORKDIR/nanvix/src/microvm
+make all
```

Comment:
this is supported from root dir

Reply:
Yes, it is. The Kernel too, but i chose this way to show that is possible.

Reply:
We should prefer building from the root directory, as build artifacts will be placed on the same output directory (ie `bin/`) as other build artifacts (ie, `kernel`).

If one builds components individually, build artifacts will be scattered across the repository. As a result, one could even end up with two binaries for `microm` (one in `bin/` and another in` src/microvm/bin`).

Could you please update instructions to guide people to *aways* build from the root directory?

Reply:
Sure. I got it.